server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -85,18 +145,70 @@ public String[] excludes() {
         return this.excludes;
     }
 
+    public Boolean includeVectors() {
+        return this.includeVectors;
+    }
+
+    public boolean filterVectorFields() {
+        return this.includeVectors == null || this.includeVectors == false;
+    }
+
```

Comment:
Because we don't want to complicate the usage? That's also a small change compared with the burden to always have to check for includes/excludes and the vector exclusion. It's also prone to error since having a vector exclusion doesn't mean that there will be any filtering (if there's no vector field in the mapping). I rely on this change in this [PR](https://github.com/elastic/elasticsearch/pull/113418) so I want to make sure that it remains simple.

Reply:
We added an `isEmpty()` method to `SourceFilter`, wouldn't it be easier to always return a non-null `SourceFilter` and have callers check that method instead?

Reply:
That would be fine yes, anything that don't require caller to be aware of the `filterVectorFields`.

Reply:
I will rework the logic to use `isEmpty` over `filterVectorFields`